service/mapper/transitions.go

```diff
@@ -358,20 +358,34 @@ func (t *Transitions) MapRegisters(s *State) error {
 	// doesn't really matter for badger if they are in random order, so this
 	// way of iterating should be fine.
 	n := 1000
-	payloads := make([]*ledger.Payload, 0, n)
-	for path, payload := range s.registers {
-		payloads = append(payloads, payload)
-		delete(s.registers, path)
-		if len(payloads) >= n {
-			break
-		}
+	if len(s.registers) < n {
+	        n = len(s.registers)
 	}
 
-	// Then we store the (maximum) 1000 paths and payloads.
-	err := t.write.Payloads(s.height, payloads)
```

Comment:
this was the firs assign to `err` now that this is gone line 393 needs to assign to `err`